pkg/cmd/stack/deploy.go

```diff
@@ -350,3 +354,40 @@ func addHiddenExposedPorts(ctx context.Context, s *model.Stack) {
 	}
 
 }
+
+func validateEndpoints(ctx context.Context, s *model.Stack, c kubernetes.Interface) error {
+
+	for endpointName, endpoint := range s.Endpoints {
+		for _, endpointRule := range endpoint.Rules {
+			svc, _ := services.Get(ctx, endpointRule.Service, s.Namespace, c)
+			if service, ok := s.Services[endpointRule.Service]; !ok && svc == nil {
+				return fmt.Errorf("Invalid endpoint '%s': service '%s' does not exist.", endpointName, endpointRule.Service)
```

Comment:
Let's no check if it exists. In this case, both stacks are deployed in parallel and the service might not be there